File: solutions/leetcode/contests/w-435/1-maximum_difference_between_even_and_odd_frequencies.java
import java.util.*;

class Solution {
    public int maxDifference(String s) {
        Map<Character, Integer> frequencyMap = new HashMap<>();
        for (char c : s.toCharArray()) {
            frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
        }
        
        List<Integer> oddFrequencies = new ArrayList<>();
        List<Integer> evenFrequencies = new ArrayList<>();
        
        for (int frequency : frequencyMap.values()) {
            if (frequency % 2 == 0) {
                evenFrequencies.add(frequency);
            } else {
                oddFrequencies.add(frequency);
            }
        }
        
        return Collections.max(oddFrequencies) - Collections.min(evenFrequencies);
    }
}

File: solutions/leetcode/contests/w-435/1-maximum_difference_between_even_and_odd_frequencies.scala
object Solution {
  def maxDifference(s: String): Int = {
    val frequencyMap = s.groupBy(identity).view.mapValues(_.length).toMap  
    val (oddFrequencies, evenFrequencies) = frequencyMap.values.partition(_ % 2 == 1)
    oddFrequencies.max - evenFrequencies.min
  }
}

